server/app/services/cloud/ApplicantFileNameFormatter.java

```diff
@@ -22,4 +22,17 @@ public static String formatFileUploadQuestionFilename(
     return String.format(
         "applicant-%d/program-%d/block-%s/${filename}", applicantId, programId, blockId);
   }
+
+  /** Check if the formatted file key matches the applicant id */
+  public static boolean isApplicantOwnedFileKey(String fileKey, long applicantId) {
+    if (fileKey.isBlank()) {
+      throw new IllegalArgumentException("'fileKey' must not be blank.");
+    }
+
+    if (applicantId <= 0) {
+      throw new IllegalArgumentException("'applicantId' must be greater than zero.");
+    }
+
+    return fileKey.contains(String.format("applicant-%d", applicantId));
```

Comment:
Nitpick (optional): Should we use `.startsWith` instead of `.contains` to have a slightly stronger verification here?

Reply:
I've seen comments indicating there was a change quite some time ago (related to the introduction of the StoredFileAcls table). Further investigation needs to be done before I'd feel comfortable making this change.

See https://github.com/civiform/civiform/blob/91efd15d5aaec0978eb51ba5cca7cec70042370b/server/app/controllers/FileController.java#L69-L77

Reply:
Ah understood, sounds good leaving as-is then!